tests/test_custom_checkers/test_one_iteration_checker.py

```diff
@@ -439,3 +440,117 @@ def test_while_with_try_except(self):
 
         with self.assertNoMessages():
             self.checker.visit_while(while_node)
+
+
+class TestOneIterationCheckerZ3Option(pylint.testutils.CheckerTestCase):
+    CHECKER_CLASS = OneIterationChecker
+    CONFIG = {"z3": True}
+
+    def test_z3_one_iteration_break_by_precondition(self):
+        src = """
+        def func(x: int) -> int:
+            '''
+            Preconditions:
+                - x > 5
+            '''
+            while x > 0:
```

Comment:
This test case is good. However, given the following variation:

```python
def func(x: int) -> int:
    """Test
    Preconditions:
        - x > 5
    """
    while x > 0:
        if x > 3:
            break
    return x
```

the `one-iteration` error is *not* reported. This suggests a bug in the algorithm for this check regarding feasible edges.